tel2puml/find_unique_graphs/otel_ingestion/ingest_otel_data.py

```diff
@@ -0,0 +1,104 @@
+"""Module to stream OTel data from a data source and store it within a
+data holder."""
+
+import yaml
+from typing import Any
+
+from tel2puml.find_unique_graphs.otel_ingestion.otel_data_holder import (
+    SQLDataHolder,
+    DataHolder,
+)
+from tel2puml.find_unique_graphs.otel_ingestion.otel_data_source import (
+    JSONDataSource,
+    OTELDataSource,
+)
+from tel2puml.find_unique_graphs.otel_ingestion.otel_data_model import (
+    SQLDataHolderConfig,
+    JSONDataSourceConfig,
+)
+
+
+class IngestData:
+    """Class to stream OTel data from a data source and store in within a data
+    holder."""
+
+    def __init__(
+        self, otel_data_source: OTELDataSource, data_holder: DataHolder
+    ) -> None:
+        """Constructor method.
+
+        :param otel_data_source: The class that returns an OTelEvent object
+        :type otel_data_source: :class: `OTELDataSource`
+        :param data_holder: The class that handles data holder operations
+        :type data_holder: `DataHolder`
+        """
+        self.data_holder = data_holder
+        self.data_source = otel_data_source
+
+    def load_to_data_holder(self) -> None:
+        """Streams OTelEvent objects to data holder."""
+
+        with self.data_holder:
+            for data in self.data_source:
+                self.data_holder.save_data(data)
+
+
+def fetch_data_source(config: Any, data_source_type: str) -> OTELDataSource:
```

Comment:
Think we should make config a typed dict. Any is slightly broad...

Reply:
Typed dict created